src/utilities/collection-utils.ts

```diff
@@ -16,10 +16,10 @@ import _ from "lodash";
  * @returns true if both arrays contain all the same elements of the other,
  *          not considering order, false otherwise.
  */
-const _equalsBy = function<T, V>(
+const _equalsBy = function <T, V>(
     selector: (element: T) => V,
-    array1: Array<T> | List<any> | undefined,
-    array2: Array<T> | List<any> | undefined
+    array1: Array<T> | Immutable.List<any> | undefined,
```

Comment:
why change this?

Reply:
also, it should be `List<T>` not `List<any>`

Reply:
In order to make this work with global distribution packages, references to any objects available from the `Immutable` library must reference the library object itself as it's named and exported from their package.  In this case, it's `Immutable`.  Technically this doesn't have to use `Immutable.List` because it's just typing the objet but there is code in this file that does reference non Typescript code (see line 90).  It's probably best practice to always use the package's main object so that packging for global distributions is never an issue.

I didn't change it to `Immutable.List<any>` because it was originally `List<any>`.

Reply:
I'm testing this out again against immutable and I think I'm wrong.  This actually does work the way it originally was. I'm going to try reverting all of these and rerunning some tests.

Reply:
@KevinBusch I thought so. Pretty sure the required `Immutable` code gets compiled into the record classes.